config/jobs/etcd-druid/druid-e2e-kind.yaml

```diff
@@ -0,0 +1,65 @@
+presubmits:
+  gardener/etcd-druid:
+    - name: pull-etcd-druid-e2e-kind
+      cluster: gardener-prow-build
+      always_run: true
+      skip_branches:
+        - release-v\d+.\d+ # don't run on release branches for now (add a job per branch later)
+      decorate: true
+      decoration_config:
+        timeout: 60m
+        grace_period: 15m
+      labels:
+        preset-dind-enabled: "true"
+        preset-kind-volume-mounts: "true"
+      annotations:
+        description: Runs KIND cluster based e2e tests for etcd druid developments in pull requests
+      spec:
+        containers:
+          - image: gcr.io/k8s-staging-test-infra/krte:v20230421-ec4335b54b-master
+            command:
+            - wrapper.sh
+            - bash
+            - -c
+            - make ci-e2e-kind
+            # we need privileged mode in order to do docker in docker
+            securityContext:
+              privileged: true
+            resources:
+              requests:
+                cpu: 12
+                memory: 18Gi
+periodics:
+  - name: ci-etcd-druid-e2e-kind
+    cluster: gardener-prow-build
+    interval: 4h
+    extra_refs:
+      - org: gardener
+        repo: etcd-druid
+        base_ref: master
+    decorate: true
+    decoration_config:
+      timeout: 60m
+      grace_period: 15m
+    labels:
+      preset-dind-enabled: "true"
+      preset-kind-volume-mounts: "true"
+    annotations:
+      description: Runs KIND cluster based e2e tests for etcd druid developments periodically
+      testgrid-dashboards: gardener-etcd-druid
+      testgrid-days-of-results: "60"
+    spec:
+      containers:
+        - image: gcr.io/k8s-staging-test-infra/v20230421-ec4335b54b-master
+          command:
+          - wrapper.sh
+          - bash
+          - -c
+          - make ci-e2e-kind
+          # we need privileged mode in order to do docker in docker
+          securityContext:
+            privileged: true
+          resources:
+            requests:
+              cpu: 12
+              memory: 18Gi
```

Comment:
One question, other than that lgtm.
Have you tested the resource consumption for this test?

Reply:
I haven't tested specifically. But that amount of resource is enough for the e2e tests. Basically, we run the tests inside a KIND cluster running inside a container.

Reply:
My only concern is not to allocate unnecessarily a lot of resources to prevent resource wastage.
/cc @oliver-goetz

Reply:
I agree to @acumino. We have nodes with 16 CPUs, so the etcd-druid test would consume an entire node.
Usually, we require 12 CPUs for our big e2e tests in `gardener/gardener` where 15 shoots are created in parallel. The smaller tests require 6 CPUs. 
Your description sound like the etcd-druid test rather fits into the small test category, so let's start with 6 CPUs.

Reply:
Okay. Let me come back to this after I check with the resources..

Reply:
@seshachalam-yv  helped me to verify that the druid e2e tests run even with 6cpu and 8Gi mem. So I changed the PR to use that much resources only.